src/features/click-to-play.js

```diff
@@ -2372,9 +2371,48 @@ const updateHandlers = {
 }
 
 export function init (args) {
+    const websiteOwner = args?.site?.parentEntity
+    const settings = args?.featureSettings?.clickToPlay || {}
+
+    for (const entity of Object.keys(config)) {
+        // Strip config entities that are first-party, or aren't enabled in the
+        // extension's clickToPlay settings.
+        // Note: To support legacy configurations consider `undefined` state as
+        //       "enabled".
+        if ((websiteOwner && entity === websiteOwner) ||
+            !settings[entity] ||
+            settings[entity].state === 'disabled') {
+            delete config[entity]
+            continue
+        }
+
+        // Populate the entities and entityData data structures.
+        // TODO: Remove them and this logic, they seem unnecessary.
+
+        entities.push(entity)
+
+        const shouldShowLoginModal = !!config[entity].informationalModal
+        const maxClicks = config[entity].clicksBeforeSimpleVersion || 3
+        const currentEntityData = { maxClicks, shouldShowLoginModal }
+
+        if (shouldShowLoginModal) {
+            const { informationalModal } = config[entity]
+            currentEntityData.modalIcon = informationalModal.icon
+            currentEntityData.modalTitle = informationalModal.messageTitle
+            currentEntityData.modalText = informationalModal.messageBody
+            currentEntityData.modalAcceptText = informationalModal.confirmButtonText
+            currentEntityData.modalRejectText = informationalModal.rejectButtonText
+        }
+
+        entityData[entity] = currentEntityData
+    }
+
+    // TODO: Fix the race condition where the getDevMode or
```

Comment:
Agree, we should have just one settings getter here.  Looks like we're already getting maxClicks above anyway; could we just pass everything thing above via init (args)?